pkg/threescale/threescale.go

```diff
@@ -284,6 +296,22 @@ func parseActionPath(path string) (string, error) {
 	return userKey, nil
 }
 
+// rpcStatusErrorHandler provides a uniform way to log and format error messages and status which should be
+// returned to the user in cases where the authorization request is rejected.
+func rpcStatusErrorHandler(userFacingErrMsg string, fn func(string) rpc.Status, err error) (rpc.Status, error) {
+	if userFacingErrMsg != "" {
+		err = fmt.Errorf("%s - %s", userFacingErrMsg, err.Error())
+	}
+
+	log.Error(err.Error())
+	return fn(err.Error()), err
+}
+
+//TODO - This should be replaced by function in istio in 1.1
+func withUnavailable(msg string) rpc.Status {
+	return status.WithMessage(14, msg)
```

Comment:
also explaining the rationale behind the magic numbers, like this `14` (or just use consts you can refer to from other places, like in the tests)

Reply:
Yeah good point. I can probably use the exposed constants.